lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -114,29 +177,85 @@ SymbolLocator *SymbolLocatorDebuginfod::CreateInstance() {
 
 static std::optional<FileSpec> GetFileForModule(
     const ModuleSpec &module_spec,
-    std::function<llvm::Expected<std::string>(llvm::object::BuildIDRef)>
-        PullFromServer) {
+    std::function<llvm::StringRef(llvm::object::BuildID)> UrlBuilder,
+    bool force_lookup) {
+  PluginProperties &plugin_props = GetGlobalPluginProperties();
+  llvm::Expected<llvm::StringRef> CacheDirectoryPathOrErr =
+      plugin_props.GetCachePath();
+  // We really do require a cache location to store stuff.
+  if (!CacheDirectoryPathOrErr)
+    return {};
+  llvm::StringRef CacheDirectoryPath = *CacheDirectoryPathOrErr;
+  llvm::SmallVector<llvm::StringRef> DebuginfodUrls =
+      llvm::getDefaultDebuginfodUrls();
+  std::chrono::milliseconds Timeout = plugin_props.GetTimeout();
   if (!ModuleList::GetGlobalModuleListProperties().GetEnableExternalLookup())
     return {};
   const UUID &module_uuid = module_spec.GetUUID();
   if (module_uuid.IsValid() && llvm::canUseDebuginfod()) {
```

Comment:
We should probably check `llvm::canUseDebuginfod() ` as soon as we enter this function and quick return if this is false. When would this return false?

The `llvm::canUseDebuginfod()` function looks like:
```
bool canUseDebuginfod() {
  return HTTPClient::isAvailable() && !getDefaultDebuginfodUrls().empty();
}
```
Is `HTTPClient::isAvailable()` super cheap to call and won't cause any latency? Is `getDefaultDebuginfodUrls()` going to have our settings?

Reply:
Yeah, it's super fast, and if someone's built without LIBCURL, it's a single (constant) instruction. This seems like a good idea.